lib/nuget-parser/nuspec-parser.ts

```diff
@@ -8,80 +8,91 @@ const debug = debugModule('snyk');
-
-              // Add the default dependencies
-              ownDeps = ownDeps.concat(
-                extractDepsFromRaw(rawDependency.dependency),
-              );
-            });
-          });
+// export async function parseNuspec(dep, targetFramework) {
+async function loadNuspecFromAsync(dep) {
+  const nupkgPath = path.resolve(
+    dep.path,
+    dep.name + '.' + dep.version + '.nupkg',
+  );
+
+  //just in case, does the code expect null to be returned on any error here?
+  if (fs.existsSync(nupkgPath) === false) return null;
+
+  const nupkgData = fs.readFileSync(nupkgPath);
+  const nuspecZipData: any = await JSZip.loadAsync(nupkgData);
+
+  const nuspecFiles = Object.keys(nuspecZipData.files).filter(file => {
+    return path.extname(file) === '.nuspec';
+  });
+
+  if (!nuspecFiles || !nuspecZipData) {
+    //sanity check, perhaps log a warning here?
+    return null;
+  }
+
+  const nuspecContent = await nuspecZipData.files[nuspecFiles[0]].async('text');
+  return nuspecContent;
+}
 
-          return resolve({
-            children: ownDeps,
-            name: dep.name,
+export async function actuallyParsedNuspec(
+  nuspecContent: any,
+  targetFramework,
+  depname,
+) {
+  const result = await parseXML.parseStringPromise(nuspecContent);
+  let ownDeps: any = [];
+  // We are only going to check the first targetFramework we encounter
+  // in the future we may want to support multiple, but only once
+  // we have dependency version conflict resolution implemented
+  if (!result.package.metadata) {
+    throw new Error(
+      'This is an invalid nuspec file. Metadata xml section is missing. This is a required element. See https://docs.microsoft.com/en-us/nuget/reference/nuspec',
+    );
+  }
+  result.package.metadata.forEach(metadata => {
+    if (metadata && metadata.dependencies) {
+      metadata.dependencies.forEach(rawDependency => {
+        // Find and add target framework version specific dependencies
+        const depsForTargetFramework = extractDepsForTargetFramework(
+          rawDependency,
+          targetFramework,
+        );
+
+        if (depsForTargetFramework && depsForTargetFramework.group) {
+          ownDeps = ownDeps.concat(
+            extractDepsFromRaw(depsForTargetFramework.group.dependency),
+          );
+        }
+
+        // Find all groups with no targetFramework attribute
+        // add their deps
+        const depsFromPlainGroups = extractDepsForPlainGroups(rawDependency);
+
+        if (depsFromPlainGroups) {
+          depsFromPlainGroups.forEach(depGroup => {
+            ownDeps = ownDeps.concat(extractDepsFromRaw(depGroup.dependency));
           });
-        });
+        }
+
+        // Add the default dependencies
+        ownDeps = ownDeps.concat(extractDepsFromRaw(rawDependency.dependency));
       });
-    })
-    .catch(err => {
-      // parsing problems are coerced into an empty nuspec
-      debug('Error parsing dependency', JSON.stringify(dep), err);
```

Comment:
This debug message was removed, please make sure we're not missing a possible error